src/org/javarosa/core/model/condition/Recalculate.java

```diff
@@ -142,14 +140,9 @@ public static IAnswerData wrapData (Object val, int dataType) {
         } else if (val instanceof String) {
             return new StringData((String)val);
         } else if (val instanceof Date) {
-            if ( dataType == Constants.DATATYPE_DATE_TIME) {
-                return new DateTimeData((Date)val);
-            }
-            else if ( dataType == Constants.DATATYPE_TIME ) {
-                return new TimeData((Date)val);
-            } else {
-                return new DateData((Date)val);
-            }
+            if (dataType == Constants.DATATYPE_TIME)
+                return new TimeData((Date) val);
+            return new DateTimeData((Date) val);
```

Comment:
Would it be better to have an explicit case for `DATATYPE_DATE`? When we know the intended time is date, I don't think we really want to add on a time of 0.

Reply:
Done!